crates/broker/src/aggregator.rs

```diff
@@ -206,10 +210,20 @@ impl AggregatorService {
                 .with_context(|| format!("Failed to get {proof_id} journal"))?
                 .with_context(|| format!("{proof_id} journal missing"))?;
 
+            let fulfillment_data = match order.request.requirements.predicate.predicateType {
+                PredicateType::ClaimDigestMatch => FulfillmentData::from_claim_digest(
+                    order.request.requirements.predicate.claim_digest().unwrap(),
+                ),
+                _ => FulfillmentData::from_image_id_and_journal(
+                    Digest::from_hex(order.image_id.unwrap()).unwrap(),
+                    journal,
+                ),
+            };
```

Comment:
nit: this could be made a method/function as we use it several times within the codebase